recipes/sqlitecpp/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class SQLiteCppConan(ConanFile):
+    name = "sqlitecpp"
+    description = "SQLiteCpp is a smart and easy to use C++ sqlite3 wrapper"
+    topics = ("conan", "sqlitecpp", "sqlite3")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/SRombauts/SQLiteCpp"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False], "lint": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "lint": False,
+                       "sqlite3:threadsafe": 2,
```

Comment:
Is this option needed?

Reply:
ping @uilianries

Reply:
@arnesor Do you know if we can use sqlite3:threadsafe=1 instead?

Reply:
Ahh okay, we need **2** because multi-thread support from sqlite: https://github.com/SRombauts/SQLiteCpp#thread-safety

We can accept with **1**, but the package will be restricted to single thread only

Few options:
- Build and offers sqlite3 with threadsafe=2
- Or, restrict SqliteCpp to single-thread by default.

Reply:
@arnesor commented here: https://github.com/bincrafters/conan-sqlitecpp/pull/3#discussion_r361749387

Reply:
Previously in bincrafters/conan-sqlite3 we solved this by having two builds of sqlite3, one with sqlite3:threadsafe=1 and one with sqlite3:threadsafe=2, see commit https://github.com/bincrafters/conan-sqlite3/commit/c46cb2882ce222c182766c4c5faae284234dbb7c

Reply:
related issue: https://github.com/conan-io/conan-center-index/issues/413

Reply:
@arnesor I'm checking VCPKG to have some idea. They don't use threadsafe option, actually, neither for sqlite3, which means, threadsafe is 0. For reference: https://github.com/microsoft/vcpkg/tree/master/ports/sqlitecpp

/cc @danimtb @SSE4

Reply:
@uilianries As I understand it SQLiteCpp supports both sqlite3 threadsafe = 0 (single thread) and threadsafe = 2 (multithread). It is just threadsafe = 1 (serialized mode) it does not support. Maybe  threadsafe = 0 could be the default for conan sqlitecpp instead of threadsafe = 2?

Reply:
@arnesor Yes, but I would like to use the safe option for both packages. Otherwise, we will need to build different packages and we don't have this support right now.

Reply:
@uilianries By "safe option", do you mean serialized mode (threadsafe = 1)? If so, how can we do that when the documentation of SQLiteCpp says that it does not support this mode?

Reply:
@arnesor No, I mean threadsafe=0

Reply:
@uilianries OK, then I agree :-)